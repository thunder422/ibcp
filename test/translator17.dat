#################################################
#  Translator Test #17: Constants (2013-09-01)  #
#################################################

A=4+B
A=4.0+B
A=4e40+B

A=B+4
A=B+4.0
A=B+4e40
A=B%+4e40

A%=4+B%
A%=4.0+B%
A%=4e40+B%

A%=B%+4
A%=B%+4.0
A%=B%+4e40

A$="4"+B$
A$=B$+"4"

A%=4\B
A%=4.0\B
A%=4e40\B

A%=B\4
A%=B\4.0
A%=B\4e40

A%=4 AND B%
A%=4.0 AND B%
A%=4e40 AND B%

A%=B% AND 4
A%=B% AND 4.0
A%=B% AND 4e40

A=- 4
A=- 4.0
A=- 4e40

A%=- 4
A%=- 4.0
A%=- 4e40

A%=NOT 4
A%=NOT 4.0
A%=NOT 4e40

A=ABS(4)
A=ABS(4.0)
A=ABS(4e40)

A$=STR$(4)
A$=STR$(4.0)
A$=STR$(4e40)

A$=CHR$(4)
A$=CHR$(4.0)
A$=CHR$(4e40)

A=4
A=4.0
A=4e40
A$="4"

A%=4
A%=4.0
A%=4e40

A=4^B
A=4.0^B
A=4e40^B

A=B^4
A=B^4.0
A=B^4e40
