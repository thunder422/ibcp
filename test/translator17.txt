
Input: A=4+B
Output: A<ref> 4 B + Assign 

Input: A=4.0+B
Output: A<ref> 4.0 B + Assign 

Input: A=4e40+B
Output: A<ref> 4e40 B + Assign 

Input: A=B+4
Output: A<ref> B 4 + Assign 

Input: A=B+4.0
Output: A<ref> B 4.0 + Assign 

Input: A=B+4e40
Output: A<ref> B 4e40 + Assign 

Input: A=B%+4e40
Output: A<ref> B% 4e40 +%1 Assign 

Input: A%=4+B%
Output: A%<ref> 4% B% +% Assign% 

Input: A%=4.0+B%
Output: A%<ref> 4.0% B% +% Assign% 

Input: A%=4e40+B%
Output: A%<ref> 4e40 B% +%2 CvtInt Assign% 

Input: A%=B%+4
Output: A%<ref> B% 4% +% Assign% 

Input: A%=B%+4.0
Output: A%<ref> B% 4.0% +% Assign% 

Input: A%=B%+4e40
Output: A%<ref> B% 4e40 +%1 CvtInt Assign% 

Input: A$="4"+B$
Output: A$<ref> "4" B$ +$ Assign$ 

Input: A$=B$+"4"
Output: A$<ref> B$ "4" +$ Assign$ 

Input: A%=4\B
Output: A%<ref> 4 B \ Assign% 

Input: A%=4.0\B
Output: A%<ref> 4.0 B \ Assign% 

Input: A%=4e40\B
Output: A%<ref> 4e40 B \ Assign% 

Input: A%=B\4
Output: A%<ref> B 4 \ Assign% 

Input: A%=B\4.0
Output: A%<ref> B 4.0 \ Assign% 

Input: A%=B\4e40
Output: A%<ref> B 4e40 \ Assign% 

Input: A%=4 AND B%
Output: A%<ref> 4% B% AND Assign% 

Input: A%=4.0 AND B%
Output: A%<ref> 4.0% B% AND Assign% 

Input: A%=4e40 AND B%
          ^^^^-- expected valid integer constant

Input: A%=B% AND 4
Output: A%<ref> B% 4% AND Assign% 

Input: A%=B% AND 4.0
Output: A%<ref> B% 4.0% AND Assign% 

Input: A%=B% AND 4e40
                 ^^^^-- expected valid integer constant

Input: A=- 4
Output: A<ref> 4% -U% CvtDbl Assign 

Input: A=- 4.0
Output: A<ref> 4.0 -U Assign 

Input: A=- 4e40
Output: A<ref> 4e40 -U Assign 

Input: A%=- 4
Output: A%<ref> 4% -U% Assign% 

Input: A%=- 4.0
Output: A%<ref> 4.0 -U CvtInt Assign% 

Input: A%=- 4e40
Output: A%<ref> 4e40 -U CvtInt Assign% 

Input: A%=NOT 4
Output: A%<ref> 4% NOT Assign% 

Input: A%=NOT 4.0
Output: A%<ref> 4.0% NOT Assign% 

Input: A%=NOT 4e40
              ^^^^-- expected valid integer constant

Input: A=ABS(4)
Output: A<ref> 4% ABS(% CvtDbl Assign 

Input: A=ABS(4.0)
Output: A<ref> 4.0 ABS( Assign 

Input: A=ABS(4e40)
Output: A<ref> 4e40 ABS( Assign 

Input: A$=STR$(4)
Output: A$<ref> 4% STR$(% Assign$ 

Input: A$=STR$(4.0)
Output: A$<ref> 4.0 STR$( Assign$ 

Input: A$=STR$(4e40)
Output: A$<ref> 4e40 STR$( Assign$ 

Input: A$=CHR$(4)
Output: A$<ref> 4% CHR$( Assign$ 

Input: A$=CHR$(4.0)
Output: A$<ref> 4.0% CHR$( Assign$ 

Input: A$=CHR$(4e40)
               ^^^^-- expected valid integer constant

Input: A=4
Output: A<ref> 4 Assign 

Input: A=4.0
Output: A<ref> 4.0 Assign 

Input: A=4e40
Output: A<ref> 4e40 Assign 

Input: A$="4"
Output: A$<ref> "4" Assign$ 

Input: A%=4
Output: A%<ref> 4% Assign% 

Input: A%=4.0
Output: A%<ref> 4.0% Assign% 

Input: A%=4e40
          ^^^^-- expected valid integer constant

Input: A=4^B
Output: A<ref> 4 B ^ Assign 

Input: A=4.0^B
Output: A<ref> 4.0 B ^ Assign 

Input: A=4e40^B
Output: A<ref> 4e40 B ^ Assign 

Input: A=B^4
Output: A<ref> B 4% ^* Assign 

Input: A=B^4.0
Output: A<ref> B 4.0 ^ Assign 

Input: A=B^4e40
Output: A<ref> B 4e40 ^ Assign 

