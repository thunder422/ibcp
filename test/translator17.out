
Input: A=4+B
Output: A = 4 + B 

Input: A=4.0+B
Output: A = 4.0 + B 

Input: A=4e40+B
Output: A = 4e40 + B 

Input: A=B+4
Output: A = B + 4 

Input: A=B+4.0
Output: A = B + 4.0 

Input: A=B+4e40
Output: A = B + 4e40 

Input: A=B%+4e40
Output: A = B% + 4e40 

Input: A%=4+B%
Output: A% = 4 + B% 

Input: A%=4.0+B%
Output: A% = 4.0 + B% 

Input: A%=4e40+B%
Output: A% = 4e40 + B% 

Input: A%=B%+4
Output: A% = B% + 4 

Input: A%=B%+4.0
Output: A% = B% + 4.0 

Input: A%=B%+4e40
Output: A% = B% + 4e40 

Input: A$="4"+B$
Output: A$ = "4" + B$ 

Input: A$=B$+"4"
Output: A$ = B$ + "4" 

Input: A%=4\B
Output: A% = 4 \ B 

Input: A%=4.0\B
Output: A% = 4.0 \ B 

Input: A%=4e40\B
Output: A% = 4e40 \ B 

Input: A%=B\4
Output: A% = B \ 4 

Input: A%=B\4.0
Output: A% = B \ 4.0 

Input: A%=B\4e40
Output: A% = B \ 4e40 

Input: A%=4 AND B%
Output: A% = 4 AND B% 

Input: A%=4.0 AND B%
Output: A% = 4.0 AND B% 

Input: A%=4e40 AND B%
          ^^^^-- expected valid integer constant

Input: A%=B% AND 4
Output: A% = B% AND 4 

Input: A%=B% AND 4.0
Output: A% = B% AND 4.0 

Input: A%=B% AND 4e40
                 ^^^^-- expected valid integer constant

Input: A=- 4
Output: A = - 4 

Input: A=- 4.0
Output: A = - 4.0 

Input: A=- 4e40
Output: A = - 4e40 

Input: A%=- 4
Output: A% = - 4 

Input: A%=- 4.0
Output: A% = - 4.0 

Input: A%=- 4e40
Output: A% = - 4e40 

Input: A%=NOT 4
Output: A% = NOT 4 

Input: A%=NOT 4.0
Output: A% = NOT 4.0 

Input: A%=NOT 4e40
              ^^^^-- expected valid integer constant

Input: A=ABS(4)
Output: A = ABS(4) 

Input: A=ABS(4.0)
Output: A = ABS(4.0) 

Input: A=ABS(4e40)
Output: A = ABS(4e40) 

Input: A$=STR$(4)
Output: A$ = STR$(4) 

Input: A$=STR$(4.0)
Output: A$ = STR$(4.0) 

Input: A$=STR$(4e40)
Output: A$ = STR$(4e40) 

Input: A$=CHR$(4)
Output: A$ = CHR$(4) 

Input: A$=CHR$(4.0)
Output: A$ = CHR$(4.0) 

Input: A$=CHR$(4e40)
               ^^^^-- expected valid integer constant

Input: A=4
Output: A = 4 

Input: A=4.0
Output: A = 4.0 

Input: A=4e40
Output: A = 4e40 

Input: A$="4"
Output: A$ = "4" 

Input: A%=4
Output: A% = 4 

Input: A%=4.0
Output: A% = 4.0 

Input: A%=4e40
          ^^^^-- expected valid integer constant

Input: A=4^B
Output: A = 4 ^ B 

Input: A=4.0^B
Output: A = 4.0 ^ B 

Input: A=4e40^B
Output: A = 4e40 ^ B 

Input: A=B^4
Output: A = B ^ 4 

Input: A=B^4.0
Output: A = B ^ 4.0 

Input: A=B^4e40
Output: A = B ^ 4e40 

